datastream-server/src/main/java/com/linkedin/datastream/server/assignment/BroadcastStrategy.java

```diff
@@ -78,7 +89,10 @@ public BroadcastStrategy(Optional<Integer> maxTasks) {
             .stream()
             .filter(x -> x.getTaskPrefix().equals(dg.getTaskPrefix()))
             .findFirst()
-            .orElse(new DatastreamTaskImpl(dg.getDatastreams()));
+            .orElse(null);
+        if (foundDatastreamTask == null) {
+          foundDatastreamTask = getOrCreateDatastreamTask(reuseTasksPerDg, dg);
+        }
```

Comment:
You can make this part of the previous expression by replacing `orElse(null)` with:
```java
.orElseGet(() -> getOrCreateDatastreamTask(reuseTasksPerDg, dg));
```

Reply:
Done.